lib/lib-storage/src/Upload.ts

```diff
@@ -96,8 +97,29 @@ export class Upload extends EventEmitter {
   async __uploadUsingPut(dataPart: RawDataPart) {
     this.isMultiPart = false;
     const params = { ...this.params, Body: dataPart.data };
-    const putResult = await this.client.send(new PutObjectCommand(params));
-    this.putResponse = putResult;
+    const [putResult, endpoint] = await Promise.all([
+      this.client.send(new PutObjectCommand(params)),
+      this.client.config.endpoint(),
+    ]);
+
+    const locationKey = this.params
+      .Key!.split("/")
+      .map((segment) => extendedEncodeURIComponent(segment))
+      .join("/");
+    const locationBucket = extendedEncodeURIComponent(this.params.Bucket!);
+
+    let Location: string;
+    if (this.client.config.forcePathStyle) {
+      Location = `${endpoint.protocol}//${endpoint.hostname}/${locationBucket}/${locationKey}`;
```

Comment:
I did some playing around with endpoint.path -> but couldn't quite figure out it's behavior, not sure if this needs adjusting.